data_service/src/match_processing/match_parser.rs

```diff
@@ -1,31 +1,151 @@
-use riven::models::match_v5::Match;
+use crate::types::{ChampionFrame, MatchTimelineFrame, PuuidToChampionMapping};
+use chrono::{DateTime, TimeDelta};
+use riven::models::match_v5::{
+    Match, MatchTimeline, MatchTimelineInfoFrameParticipantFrame, Participant,
+};
+use std::collections::HashMap;
 
-use crate::types::PuuidToChampionMapping;
+pub async fn parse_match_timeline(
+    match_timeline: MatchTimeline,
+    match_data: Match,
+) -> crate::types::MatchTimeline {
+    let match_frames = match_timeline.info.frames;
 
-pub async fn parse_match(match_data: Match) -> Vec<PuuidToChampionMapping> {
+    // Map of Participant ID to Participant object ( Found in Match Object but missing in MatchTimeline Object)
+    let participant_id_object_map: HashMap<i32, Participant> = match_data
+        .info
+        .participants
+        .iter()
+        .map(|p| (p.participant_id, p.clone()))
+        .collect();
+
+    // Participant ID to Opponent Participant ID Map
+    let participant_id_opponent_id_map: HashMap<i32, i32> = match_data
+        .info
+        .participants
+        .iter()
+        .map(|p| {
+            (
+                p.participant_id,
+                get_opponent_player_id(p.participant_id, match_data.clone()),
+            )
+        })
+        .collect();
+
+    // Using Combined context ( MatchTimeline, Match Objects ) create a new MatchTimeline Object
+    let mapped_frames = match_frames
```

Comment:
I would definitely refactor this into another private function (get_champion_frames_from_match_data, or something like that).

Reply:
makes sense. will do that.